lmcache/v1/storage_backend/connector/fs_connector.py

```diff
@@ -67,41 +72,55 @@ def _get_file_path(self, key: CacheEngineKey) -> Path:
     async def exists(self, key: CacheEngineKey) -> bool:
         """Check if key exists in file system"""
         file_path = self._get_file_path(key)
-        return file_path.exists()
+        return await aiofiles.os.path.exists(file_path)
 
     async def get(self, key: CacheEngineKey) -> Optional[MemoryObj]:
         """Get data from file system"""
         file_path = self._get_file_path(key)
-        if not file_path.exists():
-            return None
 
         try:
-            # Read file content
-            with open(file_path, "rb") as f:
-                data = f.read()
-
-            # Split metadata and actual data
-            metadata = RemoteMetadata.deserialize(memoryview(data[:METADATA_BYTES_LEN]))
-            kv_bytes = data[METADATA_BYTES_LEN : METADATA_BYTES_LEN + metadata.length]
-
-            # Allocate memory and copy data
-            memory_obj = self.local_cpu_backend.allocate(
-                metadata.shape,
-                metadata.dtype,
-                metadata.fmt,
+            t1 = time.perf_counter()
+            async with aiofiles.open(file_path, "rb") as f:
+                t2 = time.perf_counter()
+
+                # Read metadata buffer first to get shape, dtype, fmt
+                # to be able to allocate memory object for the data and read into it
+                md_buffer = bytearray(METADATA_BYTES_LEN)
+                num_read = await f.readinto(md_buffer)
+                if num_read != len(md_buffer):
+                    raise RuntimeError(
+                        f"Partial read meta {len(md_buffer)} got {num_read}"
+                    )
+                t3 = time.perf_counter()
+
+                # Deserialize metadata and allocate memory
+                metadata = RemoteMetadata.deserialize(md_buffer)
+                memory_obj = self.local_cpu_backend.allocate(
+                    metadata.shape, metadata.dtype, metadata.fmt
+                )
+                if memory_obj is None:
+                    logger.debug("Memory allocation failed during async disk load.")
+                    return None
+                t4 = time.perf_counter()
+
+                # Read the actual data into allocated memory
+                buffer = memory_obj.byte_array
+                num_read = await f.readinto(buffer)
+                if num_read != len(buffer):
+                    raise RuntimeError(
+                        f"Partial read data {len(buffer)} got {num_read}"
+                    )
+                t5 = time.perf_counter()
+
+            logger.debug(
```

Comment:
Would you mind to remove these debug purpose statements as this could introduce extra performance cost.

Reply:
I actually think it is very valuable to be able to understand how the filesystem behaves. This is only printed in DEBUG level. What actually introduces performance cost here? calling perf_counter? or the printing? because afaik both of those will be few orders of magnitude lower than any of the FS operations we do. WDYT - can we keep it?